examples_ch59x/debugprintfdemo/funconfig.h

```diff
@@ -1,10 +1,13 @@
 #ifndef _FUNCONFIG_H
 #define _FUNCONFIG_H
 
-#define FUNCONF_USE_HSI         0 // CH592 does not have HSI
-#define FUNCONF_USE_HSE         1
-#define FUNCONF_DEBUG_HARDFAULT 0
-#define FUNCONF_USE_CLK_SEC     0
-#define FUNCONF_INIT_ANALOG     0 // ADC is not implemented yet
+#define FUNCONF_USE_HSI           0 // CH592 does not have HSI
+#define FUNCONF_USE_HSE           1
+#define CLK_SOURCE_CH59X          CLK_SOURCE_PLL_60MHz // default so not really needed
+#define FUNCONF_SYSTEM_CORE_CLOCK 60 * 1000 * 1000     // keep in line with CLK_SOURCE_CH59X
```

Comment:
Same here, don't need these #define'd here.  Just #define anything that's different or needs to be #defined.

Reply:
this actually is the minimum of #defines needed. The clocks default to HSI if not given, and the others are ifdefing out unsupported features. Only the CLK_SOURCE_CH59X is not strictly needed indeed, but since it is unique to the ch59x and also quite different from the ch32's I thought it would be good to notify the end user about it.

If it's not here it definitely would have to go in the documentation i think (which will only happen in a PR further down the line)

Reply:
yes that's my vote too, I'll keep the funconfigs as they are now then.